src/transaction/transaction.rs

```diff
@@ -232,6 +234,69 @@ impl Transaction {
         .commit()
         .await
     }
+
+    /// Pessimisticly lock the keys
+    ///
+    /// ```rust,no_run
+    /// # use tikv_client::{Config, transaction::Client};
+    /// # use futures::prelude::*;
+    /// # use tikv_client_common::Key;
+    /// # futures::executor::block_on(async {
+    /// # let client = Client::new(Config::default()).await.unwrap();
+    /// let mut txn = client.begin().await.unwrap();
+    /// // ... Do some actions.
+    /// let key1: Key = b"key1".to_vec().into();
+    /// let result: () = txn.pessimistic_lock(vec![key1.clone()]).await.unwrap();
+    /// # });
+    /// ```
+    pub async fn pessimistic_lock(
+        &mut self,
+        keys: impl IntoIterator<Item = impl Into<Key>>,
+    ) -> Result<()> {
+        let mut keys: Vec<Vec<u8>> = keys
+            .into_iter()
+            .map(|it| it.into())
+            .map(|it: Key| it.into())
+            .collect();
+        keys.sort();
+        let primary_lock = keys[0].clone();
+        let lock_ttl = DEFAULT_LOCK_TTL;
+        let for_update_ts = self.rpc.clone().get_timestamp().await.unwrap().version();
+        self.for_update_ts = std::cmp::max(self.for_update_ts, for_update_ts);
+        new_pessimistic_lock_request(
+            keys,
+            primary_lock,
+            self.timestamp.version(),
+            lock_ttl,
+            for_update_ts,
+        )
+        .execute(self.rpc.clone())
+        .await
+    }
+
+    /// Commits the actions of the pessimistic transaction.
+    ///
+    /// ```rust,no_run
+    /// # use tikv_client::{Config, transaction::Client};
+    /// # use futures::prelude::*;
+    /// # futures::executor::block_on(async {
+    /// # let client = Client::new(Config::default()).await.unwrap();
+    /// let mut txn = client.begin().await.unwrap();
+    /// // ... Do some actions.
+    /// let req = txn.pessimistic_commit();
+    /// let result: () = req.await.unwrap();
+    /// # });
+    /// ```
+    pub async fn pessimistic_commit(&mut self) -> Result<()> {
```

Comment:
Similarly, we shouldn't have separate pessimistic commit API, commit should do the right thing based on whether then transaction is optimistic or pessimistic